src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -439,6 +439,9 @@ typedef struct {
   bool seen_error;
   bool exceeded_metadata_size;
 
+  /** the error that resulted in this stream being removed */
+  grpc_error *removal_error;
```

Comment:
It is used here... it's bubbled up whenever we bubble up errors in remove_stream (so it could reach the surface).

We had a problematic case whereby we weren't saving this away and were then forced to synthesize it (around line 1812) - this created two errors per request on a common code path, and accounted for ~4us on the trace I was looking at.